sycl/include/CL/sycl/detail/spirv.hpp

```diff
@@ -0,0 +1,51 @@
+//===-- spirv.hpp - Helpers to generate SPIR-V instructions ----*- C++ -*--===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+#include <CL/__spirv/spirv_ops.hpp>
+#include <CL/__spirv/spirv_types.hpp>
+#include <CL/__spirv/spirv_vars.hpp>
+#include <CL/sycl/detail/generic_type_traits.hpp>
+#include <CL/sycl/detail/type_traits.hpp>
+
+#ifdef __SYCL_DEVICE_ONLY__
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+namespace spirv {
+template <__spv::Scope S, typename T, typename IdT>
```

Comment:
Probably worth adding a comment clarifying that first template handles scalar local index, whereas second is for vector variant.
It takes some time to understand the difference as both templates have the same name.